src/mono/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -1754,7 +1751,7 @@ public static ConstructorInfo GetConstructor(Type type, ConstructorInfo construc
 
             ConstructorInfo res = type.GetConstructor(constructor);
             if (res == null)
-                throw new ArgumentException("constructor not found");
+                throw new ArgumentException(SR.Argument_ConstructorNotFound);
```

Comment:
`type.GetConstructor` should not ever return null (the nullable annotations say that it does not return null and the implementation seems to match it).

This check and error message should be deleted or changed to assert.